subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/inputs/undeclared/UndeclaredFileInputsIntegrationTest.groovy

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.configurationcache.inputs.undeclared
+
+import groovy.test.NotYetImplemented
+import org.gradle.configurationcache.AbstractConfigurationCacheIntegrationTest
+import org.gradle.test.fixtures.file.TestFile
+
+import static org.gradle.configurationcache.inputs.undeclared.FileUtils.testFilePath
+import static org.hamcrest.CoreMatchers.containsString
+import static org.hamcrest.CoreMatchers.startsWith
+import static org.hamcrest.Matchers.allOf
+
+class UndeclaredFileInputsIntegrationTest extends AbstractConfigurationCacheIntegrationTest implements GroovyPluginImplementation {
+    @NotYetImplemented
```

Comment:
I think we prefer to use `@ToBeImplemented` instead to test for a specific kind of failure.